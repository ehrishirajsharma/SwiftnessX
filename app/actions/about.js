// @flow
export const OPEN_ABOUT = 'OPEN_ABOUT';
export const CLOSE_ABOUT = 'CLOSE_ABOUT';

export function openAbout() {
  return {
    type: OPEN_ABOUT
  };
}

export function closeAbout() {
  return {
    type: CLOSE_ABOUT
  };
}
