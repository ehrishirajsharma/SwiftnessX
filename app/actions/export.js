// @flow
export const OPEN_EXPORT = 'OPEN_EXPORT';
export const CLOSE_EXPORT = 'CLOSE_EXPORT';

export function openExport() {
  return {
    type: OPEN_EXPORT
  };
}

export function closeExport() {
  return {
    type: CLOSE_EXPORT
  };
}
