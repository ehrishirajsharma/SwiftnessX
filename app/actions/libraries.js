// @flow
import uuidv4 from 'uuid/v4';
import { libraryType } from '../reducers/libraries';

export const ADD_LIBRARY = 'ADD_LIBRARY';
export const ADD_LIBRARY_FOLDER = 'ADD_LIBRARY_FOLDER';
export const ADD_LIBRARY_ITEM = 'ADD_LIBRARY_ITEM';

export const SAVE_LIBRARY = 'SAVE_LIBRARY';
export const SAVE_LIBRARY_FOLDER = 'SAVE_LIBRARY_FOLDER';
export const SAVE_LIBRARY_ITEM = 'SAVE_LIBRARY_ITEM';

export const EDIT_LIBRARY_TITLE = 'EDIT_LIBRARY_TITLE';
export const EDIT_LIBRARY_ORDER = 'EDIT_LIBRARY_ORDER';
export const EDIT_LIBRARY_COLOR = 'EDIT_LIBRARY_COLOR';

export const EDIT_LIBRARY_FOLDER_TITLE = 'EDIT_LIBRARY_FOLDER_TITLE';
export const EDIT_LIBRARY_FOLDER_ORDER = 'EDIT_LIBRARY_FOLDER_ORDER';

export const EDIT_LIBRARY_ITEM_TITLE = 'EDIT_LIBRARY_ITEM_TITLE';
export const EDIT_LIBRARY_ITEM_CONTENT = 'EDIT_LIBRARY_ITEM_CONTENT';
export const EDIT_LIBRARY_ITEM_ORDER = 'EDIT_LIBRARY_ITEM_ORDER';
export const EDIT_LIBRARY_ITEM_COLOR = 'EDIT_LIBRARY_ITEM_COLOR';

export const REMOVE_LIBRARY = 'REMOVE_LIBRARY';
export const REMOVE_LIBRARY_FOLDER = 'REMOVE_LIBRARY_FOLDER';
export const REMOVE_LIBRARY_ITEM = 'REMOVE_LIBRARY_ITEM';

export const IMPORT_LIBRARIES = 'IMPORT_LIBRARIES';

export function addLibrary() {
  return {
    type: ADD_LIBRARY,
    payload: {
      id: uuidv4(),
      folderId: uuidv4()
    }
  };
}

export function addLibraryFolder(id: string) {
  return {
    type: ADD_LIBRARY_FOLDER,
    payload: {
      id,
      folderId: uuidv4()
    }
  };
}

export function addLibraryItem(id: string, folderId: string) {
  return {
    type: ADD_LIBRARY_ITEM,
    payload: {
      id,
      folderId,
      itemId: uuidv4()
    }
  };
}

export function saveLibrary(id: string) {
  return {
    type: SAVE_LIBRARY,
    payload: {
      id
    }
  };
}

export function saveLibraryFolder(id: string, folderId: string) {
  return {
    type: SAVE_LIBRARY_FOLDER,
    payload: {
      id,
      folderId
    }
  };
}

export function saveLibraryItem(id: string, folderId: string, itemId: string) {
  return {
    type: SAVE_LIBRARY_ITEM,
    payload: {
      id,
      folderId,
      itemId
    }
  };
}

export function editLibraryTitle(id: string, title: string) {
  return {
    type: EDIT_LIBRARY_TITLE,
    payload: {
      id,
      title
    }
  };
}

export function editLibraryOrder(fromIndex: number, toIndex: number) {
  return {
    type: EDIT_LIBRARY_ORDER,
    payload: {
      fromIndex,
      toIndex
    }
  };
}

export function editLibraryColor(id: string, color: string) {
  return {
    type: EDIT_LIBRARY_COLOR,
    payload: {
      id,
      color
    }
  };
}

export function editLibraryFolderTitle(
  id: string,
  folderId: string,
  title: string
) {
  return {
    type: EDIT_LIBRARY_FOLDER_TITLE,
    payload: {
      id,
      folderId,
      title
    }
  };
}

export function editLibraryFolderOrder(
  id: string,
  fromIndex: number,
  toIndex: number
) {
  return {
    type: EDIT_LIBRARY_FOLDER_ORDER,
    payload: {
      id,
      fromIndex,
      toIndex
    }
  };
}

export function editLibraryItemTitle(
  id: string,
  folderId: string,
  itemId: string,
  title: string
) {
  return {
    type: EDIT_LIBRARY_ITEM_TITLE,
    payload: {
      id,
      folderId,
      itemId,
      title
    }
  };
}

export function editLibraryItemContent(
  id: string,
  folderId: string,
  itemId,
  content: string
) {
  return {
    type: EDIT_LIBRARY_ITEM_CONTENT,
    payload: {
      id,
      folderId,
      itemId,
      content
    }
  };
}

export function editLibraryItemOrder(
  id: string,
  folderId: string,
  fromIndex: number,
  toIndex: number
) {
  return {
    type: EDIT_LIBRARY_ITEM_ORDER,
    payload: {
      id,
      folderId,
      fromIndex,
      toIndex
    }
  };
}

export function editLibraryItemColor(
  id: string,
  folderId: string,
  itemId: string,
  color: string
) {
  return {
    type: EDIT_LIBRARY_ITEM_COLOR,
    payload: {
      id,
      folderId,
      itemId,
      color
    }
  };
}

export function removeLibrary(id: string) {
  return {
    type: REMOVE_LIBRARY,
    payload: {
      id
    }
  };
}

export function removeLibraryFolder(id: string, folderId: string) {
  return {
    type: REMOVE_LIBRARY_FOLDER,
    payload: {
      id,
      folderId
    }
  };
}

export function removeLibraryItem(
  id: string,
  folderId: string,
  itemId: string
) {
  return {
    type: REMOVE_LIBRARY_ITEM,
    payload: {
      id,
      folderId,
      itemId
    }
  };
}

export function importLibraries(libraries: libraryType[]) {
  return {
    type: IMPORT_LIBRARIES,
    payload: {
      libraries: regenerateIds(libraries)
    }
  };
}

function regenerateIds(libraries: libraryType[]) {
  return libraries.map(library => {
    const newLibrary = library;

    newLibrary.id = uuidv4();
    newLibrary.folders = library.folders.map(folder => {
      const newFolder = folder;

      newFolder.id = uuidv4();
      newFolder.checklist = folder.checklist.map(item => {
        const newItem = item;

        newItem.id = uuidv4();

        return newItem;
      });

      return newFolder;
    });

    return newLibrary;
  });
}
