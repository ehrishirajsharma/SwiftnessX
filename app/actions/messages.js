// @flow
export const DO_NOT_SHOW_DELETE_CONFIRMATION =
  'DO_NOT_SHOW_DELETE_CONFIRMATION';

export function doNotShowDeleteConfirmation() {
  return {
    type: DO_NOT_SHOW_DELETE_CONFIRMATION
  };
}
