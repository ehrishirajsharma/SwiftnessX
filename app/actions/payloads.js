// @flow
import uuidv4 from 'uuid/v4';
import { payloadType } from '../reducers/payloads';

export const ADD_PAYLOAD_ITEM = 'ADD_PAYLOAD_ITEM';
export const ADD_PAYLOAD_ITEM_DATA = 'ADD_PAYLOAD_ITEM_DATA';

export const SAVE_PAYLOAD_ITEM = 'SAVE_PAYLOAD_ITEM';

export const EDIT_PAYLOAD_ITEM_TITLE = 'EDIT_PAYLOAD_ITEM_TITLE';
export const EDIT_PAYLOAD_ITEM_ORDER = 'EDIT_PAYLOAD_ITEM_ORDER';

export const REMOVE_PAYLOAD_ITEM = 'REMOVE_PAYLOAD_ITEM';
export const REMOVE_PAYLOAD_ITEM_DATA = 'REMOVE_PAYLOAD_ITEM_DATA';

export const IMPORT_PAYLOADS = 'IMPORT_PAYLOADS';

export function addPayloadItem() {
  return {
    type: ADD_PAYLOAD_ITEM,
    payload: {
      id: uuidv4()
    }
  };
}

export function addPayloadItemData(
  id: string,
  title: string,
  description: string,
  content: string
) {
  return {
    type: ADD_PAYLOAD_ITEM_DATA,
    payload: {
      id,
      itemId: uuidv4(),
      title,
      description,
      content
    }
  };
}

export function savePayloadItem(id: string) {
  return {
    type: SAVE_PAYLOAD_ITEM,
    payload: {
      id
    }
  };
}

export function editPayloadItemTitle(id: string, title: string) {
  return {
    type: EDIT_PAYLOAD_ITEM_TITLE,
    payload: {
      id,
      title
    }
  };
}

export function editPayloadItemOrder(fromIndex: number, toIndex: number) {
  return {
    type: EDIT_PAYLOAD_ITEM_ORDER,
    payload: {
      fromIndex,
      toIndex
    }
  };
}

export function removePayloadItem(id: string) {
  return {
    type: REMOVE_PAYLOAD_ITEM,
    payload: {
      id
    }
  };
}

export function removePayloadItemData(id: string, itemId: string) {
  return {
    type: REMOVE_PAYLOAD_ITEM_DATA,
    payload: {
      id,
      itemId
    }
  };
}

export function importPayloads(payloads: payloadType[]) {
  return {
    type: IMPORT_PAYLOADS,
    payload: {
      payloads: regenerateIds(payloads)
    }
  };
}

function regenerateIds(payloads: payloadType[]) {
  return payloads.map(payload => {
    const newPayload = { ...payload };

    newPayload.id = uuidv4();

    return newPayload;
  });
}
