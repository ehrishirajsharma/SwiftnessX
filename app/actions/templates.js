// @flow
import uuidv4 from 'uuid/v4';
import { templateType } from '../reducers/templates';

export const ADD_TEMPLATE_ITEM = 'ADD_TEMPLATE_ITEM';
export const SAVE_TEMPLATE_ITEM = 'SAVE_TEMPLATE_ITEM';
export const EDIT_TEMPLATE_ITEM_TITLE = 'EDIT_TEMPLATE_ITEM_TITLE';
export const EDIT_TEMPLATE_ITEM_CONTENT = 'EDIT_TEMPLATE_ITEM_CONTENT';
export const EDIT_TEMPLATE_ITEM_ORDER = 'EDIT_TEMPLATE_ITEM_ORDER';
export const EDIT_TEMPLATE_ITEM_COLOR = 'EDIT_TEMPLATE_ITEM_COLOR';
export const REMOVE_TEMPLATE_ITEM = 'REMOVE_TEMPLATE_ITEM';

export const IMPORT_TEMPLATES = 'IMPORT_TEMPLATES';

export function addTemplateItem() {
  return {
    type: ADD_TEMPLATE_ITEM,
    payload: {
      id: uuidv4()
    }
  };
}

export function saveTemplateItem(id: string) {
  return {
    type: SAVE_TEMPLATE_ITEM,
    payload: {
      id
    }
  };
}

export function editTemplateItemTitle(id: string, title: string) {
  return {
    type: EDIT_TEMPLATE_ITEM_TITLE,
    payload: {
      id,
      title
    }
  };
}

export function editTemplateItemContent(id: string, content: string) {
  return {
    type: EDIT_TEMPLATE_ITEM_CONTENT,
    payload: {
      id,
      content
    }
  };
}

export function editTemplateItemOrder(fromIndex: number, toIndex: number) {
  return {
    type: EDIT_TEMPLATE_ITEM_ORDER,
    payload: {
      fromIndex,
      toIndex
    }
  };
}

export function editTemplateItemColor(id: string, color: string) {
  return {
    type: EDIT_TEMPLATE_ITEM_COLOR,
    payload: {
      id,
      color
    }
  };
}

export function removeTemplateItem(id: string) {
  return {
    type: REMOVE_TEMPLATE_ITEM,
    payload: {
      id
    }
  };
}

export function importTemplates(templates: templateType[]) {
  return {
    type: IMPORT_TEMPLATES,
    payload: {
      templates: regenerateIds(templates)
    }
  };
}

function regenerateIds(templates: templateType[]) {
  return templates.map(template => {
    const newTemplate = { ...template };

    newTemplate.id = uuidv4();

    return newTemplate;
  });
}
