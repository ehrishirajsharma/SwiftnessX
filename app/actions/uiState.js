// @flow
export const OPEN_TARGET = 'OPEN_TARGET';
export const OPEN_LIBRARY = 'OPEN_LIBRARY';
export const OPEN_TEMPLATE = 'OPEN_TEMPLATE';
export const OPEN_PAYLOAD = 'OPEN_PAYLOAD';
export const OPEN_CHECKLIST_DATA = 'OPEN_CHECKLIST_DATA';
export const OPEN_NOTE_DATA = 'OPEN_NOTE_DATA';
export const OPEN_LIBRARY_DATA = 'OPEN_LIBRARY_DATA';
export const OPEN_TEMPLATE_DATA = 'OPEN_TEMPLATE_DATA';
export const OPEN_PAYLOAD_DATA = 'OPEN_PAYLOAD_DATA';

export const CLOSE_LIST = 'CLOSE_LIST';
export const CLOSE_ITEM_DATA = 'CLOSE_ITEM_DATA';

export const SEARCH = 'SEARCH';

export const TOGGLE_MENU_COLOR = 'TOGGLE_MENU_COLOR';
export const TOGGLE_MAIN_COLOR = 'TOGGLE_MAIN_COLOR';

export function openTarget(id: string, folderId: string) {
  return {
    type: OPEN_TARGET,
    payload: {
      id,
      folderId
    }
  };
}

export function openLibrary(id: string, folderId: string) {
  return {
    type: OPEN_LIBRARY,
    payload: {
      id,
      folderId
    }
  };
}

export function openTemplate() {
  return {
    type: OPEN_TEMPLATE
  };
}

export function openPayload() {
  return {
    type: OPEN_PAYLOAD
  };
}

export function openChecklistData(id?: string, auto?: boolean = false) {
  return {
    type: OPEN_CHECKLIST_DATA,
    payload: {
      id,
      auto
    }
  };
}

export function openNoteData(id?: string) {
  return {
    type: OPEN_NOTE_DATA,
    payload: {
      id
    }
  };
}

export function openLibraryData(id?: string, auto?: boolean = false) {
  return {
    type: OPEN_LIBRARY_DATA,
    payload: {
      id,
      auto
    }
  };
}

export function openTemplateData(id?: string) {
  return {
    type: OPEN_TEMPLATE_DATA,
    payload: {
      id
    }
  };
}

export function openPayloadData(id?: string) {
  return {
    type: OPEN_PAYLOAD_DATA,
    payload: {
      id
    }
  };
}

export function closeList() {
  return {
    type: CLOSE_LIST
  };
}

export function closeItemData() {
  return {
    type: CLOSE_ITEM_DATA
  };
}

export function search(query: string) {
  return {
    type: SEARCH,
    payload: {
      query
    }
  };
}

export function toggleMenuColor(color: string) {
  return {
    type: TOGGLE_MENU_COLOR,
    payload: {
      color
    }
  };
}

export function toggleMainColor(color: string) {
  return {
    type: TOGGLE_MAIN_COLOR,
    payload: {
      color
    }
  };
}
