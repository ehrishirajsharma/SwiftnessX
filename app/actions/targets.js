// @flow
import uuidv4 from 'uuid/v4';
import { targetType } from '../reducers/targets';
import { libraryType } from '../reducers/libraries';

export const ADD_TARGET = 'ADD_TARGET';
export const ADD_TARGET_FOLDER = 'ADD_TARGET_FOLDER';
export const ADD_TARGET_CHECKITEM = 'ADD_TARGET_CHECKITEM';
export const ADD_TARGET_NOTE = 'ADD_TARGET_NOTE';

export const SAVE_TARGET = 'SAVE_TARGET';
export const SAVE_TARGET_FOLDER = 'SAVE_TARGET_FOLDER';
export const SAVE_TARGET_CHECKITEM = 'SAVE_TARGET_CHECKITEM';
export const SAVE_TARGET_NOTE = 'SAVE_TARGET_NOTE';

export const EDIT_TARGET_TITLE = 'EDIT_TARGET_TITLE';
export const EDIT_TARGET_ORDER = 'EDIT_TARGET_ORDER';

export const EDIT_TARGET_FOLDER_TITLE = 'EDIT_TARGET_FOLDER_TITLE';
export const EDIT_TARGET_FOLDER_ORDER = 'EDIT_TARGET_FOLDER_ORDER';
export const EDIT_TARGET_FOLDER_COLOR = 'EDIT_TARGET_FOLDER_COLOR';

export const EDIT_TARGET_CHECKITEM_TITLE = 'EDIT_TARGET_CHECKITEM_TITLE';
export const EDIT_TARGET_CHECKITEM_CONTENT = 'EDIT_TARGET_CHECKITEM_CONTENT';
export const EDIT_TARGET_CHECKITEM_ORDER = 'EDIT_TARGET_CHECKITEM_ORDER';
export const EDIT_TARGET_CHECKITEM_CHECKBOX = 'EDIT_TARGET_CHECKITEM_CHECKBOX';
export const EDIT_TARGET_CHECKITEM_COLOR = 'EDIT_TARGET_CHECKITEM_COLOR';

export const EDIT_TARGET_NOTE_TITLE = 'EDIT_TARGET_NOTE_TITLE';
export const EDIT_TARGET_NOTE_CONTENT = 'EDIT_TARGET_NOTE_CONTENT';
export const EDIT_TARGET_NOTE_ORDER = 'EDIT_TARGET_NOTE_ORDER';
export const EDIT_TARGET_NOTE_COLOR = 'EDIT_TARGET_NOTE_COLOR';

export const REMOVE_TARGET = 'REMOVE_TARGET';
export const REMOVE_TARGET_FOLDER = 'REMOVE_TARGET_FOLDER';
export const REMOVE_TARGET_CHECKITEM = 'REMOVE_TARGET_CHECKITEM';
export const REMOVE_TARGET_NOTE = 'REMOVE_TARGET_NOTE';

export const IMPORT_TARGETS = 'IMPORT_TARGETS';

export function addTarget(library?: libraryType) {
  return {
    type: ADD_TARGET,
    payload: {
      id: uuidv4(),
      folderId: uuidv4(),
      library: library ? regenerateItem(library) : undefined
    }
  };
}

export function addTargetFolder(id: string) {
  return {
    type: ADD_TARGET_FOLDER,
    payload: {
      id,
      folderId: uuidv4()
    }
  };
}

export function addTargetCheckitem(id: string, folderId: string) {
  return {
    type: ADD_TARGET_CHECKITEM,
    payload: {
      id,
      folderId,
      checkItemId: uuidv4()
    }
  };
}

export function addTargetNote(id: string, folderId: string) {
  return {
    type: ADD_TARGET_NOTE,
    payload: {
      id,
      folderId,
      noteId: uuidv4()
    }
  };
}

export function saveTarget(id: string) {
  return {
    type: SAVE_TARGET,
    payload: {
      id
    }
  };
}

export function saveTargetFolder(id: string, folderId: string) {
  return {
    type: SAVE_TARGET_FOLDER,
    payload: {
      id,
      folderId
    }
  };
}

export function saveTargetCheckitem(
  id: string,
  folderId: string,
  itemId: string
) {
  return {
    type: SAVE_TARGET_CHECKITEM,
    payload: {
      id,
      folderId,
      itemId
    }
  };
}

export function saveTargetNote(id: string, folderId: string, itemId: string) {
  return {
    type: SAVE_TARGET_NOTE,
    payload: {
      id,
      folderId,
      itemId
    }
  };
}

export function editTargetTitle(id: string, title: string) {
  return {
    type: EDIT_TARGET_TITLE,
    payload: {
      id,
      title
    }
  };
}

export function editTargetOrder(fromIndex: number, toIndex: number) {
  return {
    type: EDIT_TARGET_ORDER,
    payload: {
      fromIndex,
      toIndex
    }
  };
}

export function editTargetFolderTitle(
  id: string,
  folderId: string,
  title: string
) {
  return {
    type: EDIT_TARGET_FOLDER_TITLE,
    payload: {
      id,
      folderId,
      title
    }
  };
}

export function editTargetFolderOrder(
  id: string,
  fromIndex: number,
  toIndex: number
) {
  return {
    type: EDIT_TARGET_FOLDER_ORDER,
    payload: {
      id,
      fromIndex,
      toIndex
    }
  };
}

export function editTargetFolderColor(
  id: string,
  folderId: string,
  color: string
) {
  return {
    type: EDIT_TARGET_FOLDER_COLOR,
    payload: {
      id,
      folderId,
      color
    }
  };
}

export function editTargetCheckitemTitle(
  id: string,
  folderId: string,
  itemId: string,
  title: string
) {
  return {
    type: EDIT_TARGET_CHECKITEM_TITLE,
    payload: {
      id,
      folderId,
      itemId,
      title
    }
  };
}

export function editTargetCheckitemContent(
  id: string,
  folderId: string,
  itemId: string,
  content: string
) {
  return {
    type: EDIT_TARGET_CHECKITEM_CONTENT,
    payload: {
      id,
      folderId,
      itemId,
      content
    }
  };
}

export function editTargetCheckitemOrder(
  id: string,
  folderId: string,
  fromIndex: number,
  toIndex: number
) {
  return {
    type: EDIT_TARGET_CHECKITEM_ORDER,
    payload: {
      id,
      folderId,
      fromIndex,
      toIndex
    }
  };
}

export function editTargetCheckitemCheckbox(
  id: string,
  folderId: string,
  itemId: string,
  checked: boolean
) {
  return {
    type: EDIT_TARGET_CHECKITEM_CHECKBOX,
    payload: {
      id,
      folderId,
      itemId,
      checked
    }
  };
}

export function editTargetCheckitemColor(
  id: string,
  folderId: string,
  itemId: string,
  color: string
) {
  return {
    type: EDIT_TARGET_CHECKITEM_COLOR,
    payload: {
      id,
      folderId,
      itemId,
      color
    }
  };
}

export function editTargetNoteTitle(
  id: string,
  folderId: string,
  itemId: string,
  title: string
) {
  return {
    type: EDIT_TARGET_NOTE_TITLE,
    payload: {
      id,
      folderId,
      itemId,
      title
    }
  };
}

export function editTargetNoteContent(
  id: string,
  folderId: string,
  itemId: string,
  content: string
) {
  return {
    type: EDIT_TARGET_NOTE_CONTENT,
    payload: {
      id,
      folderId,
      itemId,
      content
    }
  };
}

export function editTargetNoteOrder(
  id: string,
  folderId: string,
  fromIndex: number,
  toIndex: number
) {
  return {
    type: EDIT_TARGET_NOTE_ORDER,
    payload: {
      id,
      folderId,
      fromIndex,
      toIndex
    }
  };
}

export function editTargetNoteColor(
  id: string,
  folderId: string,
  itemId: string,
  color: string
) {
  return {
    type: EDIT_TARGET_NOTE_COLOR,
    payload: {
      id,
      folderId,
      itemId,
      color
    }
  };
}

export function removeTarget(id: string) {
  return {
    type: REMOVE_TARGET,
    payload: {
      id
    }
  };
}

export function removeTargetFolder(id: string, folderId: string) {
  return {
    type: REMOVE_TARGET_FOLDER,
    payload: {
      id,
      folderId
    }
  };
}

export function removeTargetCheckitem(
  id: string,
  folderId: string,
  itemId: string
) {
  return {
    type: REMOVE_TARGET_CHECKITEM,
    payload: {
      id,
      folderId,
      itemId
    }
  };
}

export function removeTargetNote(id: string, folderId: string, itemId: string) {
  return {
    type: REMOVE_TARGET_NOTE,
    payload: {
      id,
      folderId,
      itemId
    }
  };
}

export function importTargets(targets: targetType[]) {
  return {
    type: IMPORT_TARGETS,
    payload: {
      targets: regenerateItems(targets)
    }
  };
}

function regenerateItems(targets: targetType[]) {
  return targets.map(regenerateItem);
}

function regenerateItem(target: targetType) {
  const newTarget = { ...target };

  newTarget.id = uuidv4();
  newTarget.folders = target.folders.map(folder => {
    const newFolder = { ...folder };

    newFolder.id = uuidv4();

    const { checklist } = folder;
    if (checklist) {
      newFolder.checklist = checklist.map(item => {
        const newItem = { ...item };

        newItem.id = uuidv4();

        return newItem;
      });
    }

    const { notes } = folder;
    if (notes) {
      newFolder.notes = notes.map(item => {
        const newItem = { ...item };

        newItem.id = uuidv4();

        return newItem;
      });
    }

    return newFolder;
  });

  return newTarget;
}
