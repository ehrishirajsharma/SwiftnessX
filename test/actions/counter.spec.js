import { spy } from 'sinon';
import * as actions from '../../app/actions/counter';

describe('actions', () => {
  it('should increment should create increment action', () => {
    expect(actions.increment()).toMatchSnapshot();
  });

  it('should decrement should create decrement action', () => {
    expect(actions.decrement()).toMatchSnapshot();
  });

  it('should incrementIfOdd should create increment action', () => {
    const fn = actions.incrementIfOdd();
    expect(fn).toBeInstanceOf(Function);
    const dispatch = spy();
    const getState = () => ({ counter: 1 });
    fn(dispatch, getState);
    expect(dispatch.calledWith({ type: actions.INCREMENT_COUNTER })).toBe(true);
  });

  it('should incrementIfOdd shouldnt create increment action if counter is even', () => {
    const fn = actions.incrementIfOdd();
    const dispatch = spy();
    const getState = () => ({ counter: 2 });
    fn(dispatch, getState);
    expect(dispatch.called).toBe(false);
  });

  // There's no nice way to test this at the moment...
  it('should incrementAsync', done => {
    const fn = actions.incrementAsync(1);
    expect(fn).toBeInstanceOf(Function);
    const dispatch = spy();
    fn(dispatch);
    setTimeout(() => {
      expect(dispatch.calledWith({ type: actions.INCREMENT_COUNTER })).toBe(
        true
      );
      done();
    }, 5);
  });
});
