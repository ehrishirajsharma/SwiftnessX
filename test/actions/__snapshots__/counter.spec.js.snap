// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should decrement should create decrement action 1`] = `
Object {
  "type": "DECREMENT_COUNTER",
}
`;

exports[`actions should increment should create increment action 1`] = `
Object {
  "type": "INCREMENT_COUNTER",
}
`;
